pkgs/applications/science/logic/protege/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv
+, fetchzip
+, jre
+, makeWrapper
+, makeDesktopItem
+, iconConvTools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "protege";
+  version = "5.5.0";
+
+  src = fetchzip {
+    url = "https://github.com/protegeproject/protege-distribution/releases/download/v${version}/Protege-${version}-platform-independent.zip";
+    sha256 = "1v82ph1pqvnc1qynhiapzw0jwm9rphsb580lc96zwsvhrr0wd690";
+  };
+
+  nativeBuildInputs = [ makeWrapper iconConvTools ];
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    desktopName = "Protege";
+    genericName = meta.description;
+    comment = meta.description;
+    categories = "Development;";
+    icon = pname;
+    extraEntries = ''
+      StartupWMClass=${pname}
+    '';
+  };
+
+  # makeWrapper flags adapted from protege's run.sh script (see extracted sources)
+  installPhase = ''
+    mkdir -p $out/src
+    mv * $out/src
+
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/${pname} \
```

Comment:
This wrapper is a bit problematic as it reproduces the included `run.sh` file which then needs to be kept in sync. I patched the included script instead in my derivation
```
substituteInPlace $out/run.sh --replace 'java ' '${jre8}/bin/java ' --replace 'bundles' "$out/bundles" --replace ':bin' ":$out/bin"
chmod +x $out/run.sh
ln -s $out/run.sh $out/bin/protege
```

Reply:
Nice. Would you mind sharing your derivation?

Reply:
@JonathanReeve I could share it but it basically is the same as this derivation with the changed `installPhase` as described above

Reply:
Thanks @mgttlinger, I've adapted your solution with a few modifications and seems to work great!